src/main/java/org/takes/facets/auth/codecs/CcGzip.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.facets.auth.codecs;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+import lombok.EqualsAndHashCode;
+import org.takes.facets.auth.Identity;
+
+/**
+ * Gzip codec.
+ *
+ * <p>The class is immutable and thread-safe.
+ *
+ * @author Aleksey Kurochka (eg04lt3r@gmail.com)
+ * @version $Id$
+ * @since 0.15.1
+ */
+@EqualsAndHashCode(of = "origin")
+public final class CcGzip implements Codec {
+
+    /**
+     * Original codec.
+     */
+    private final transient Codec origin;
+
+    /**
+     * Ctor.
+     * @param codec Original
+     */
+    public CcGzip(final Codec codec) {
+        this.origin = codec;
+    }
+
+    @Override
+    public byte[] encode(final Identity identity) throws IOException {
+        return this.compress(this.origin.encode(identity));
+    }
+
+    @Override
+    public Identity decode(final byte[] bytes) throws IOException {
+        return this.origin.decode(this.uncompress(bytes));
+    }
+
+    /**
+     * Compress array of bytes.
+     * @param input The input to compress
+     * @return Compressed output
+     * @throws IOException If fails
+     */
+    private byte[] compress(final byte[] input) throws IOException {
```

Comment:
@eg04lt3r Better inline this method with `encode` above, it's not used anywhere else.